src/thunderbots/software/grsim_communication/motion_controller.cpp

```diff
@@ -39,19 +46,52 @@ MotionController::Velocity MotionController::bangBangVelocityController(
     }
     else
     {
-        robot_velocities.linear_velocity = MotionController::determineLinearVelocity(
-            robot, dest, desired_final_speed, delta_time, max_speed_meters_per_second,
-            max_angular_speed_radians_per_second);
-        robot_velocities.angular_velocity = MotionController::determineAngularVelocity(
-            robot, desired_final_orientation, delta_time,
-            max_angular_speed_radians_per_second,
-            max_angular_acceleration_meters_per_second_squared);
+        std::visit(
+            overload{
+                [&robot_velocities, robot, delta_time,
+                 this](MotionController::PositionCommand& command) {
+                    robot_velocities.linear_velocity =
+                        MotionController::determineLinearVelocityFromPosition(
+                            robot, command.global_destination,
+                            command.final_speed_at_destination, delta_time,
+                            this->max_speed_meters_per_second,
+                            this->max_angular_speed_radians_per_second);
+                    robot_velocities.angular_velocity =
+                        MotionController::determineAngularVelocityFromPosition(
+                            robot, command.final_orientation, delta_time,
+                            this->max_angular_speed_radians_per_second,
+                            this->max_angular_acceleration_meters_per_second_squared);
+                },
+
+                [&robot_velocities, robot, delta_time,
+                 this](MotionController::VelocityCommand& command) {
+                    if (command.linear_velocity.len() > this->max_speed_meters_per_second)
```

Comment:
I don't think we should `throw` anything here. We can be more robust as simply do the "best we can", and ultimately cap the robot at the max allowed speed.